docker.sh

```diff
@@ -1,39 +1,55 @@
 #!/bin/bash
-set -eo pipefail
+set -euox pipefail
 
-PARAM=${1:-push};
+NAME="indexd"
+PARAM=${1:-DO_NOT_PUSH};
 
-IMAGE_NAME="${DOCKER_RELEASE_REGISTRY:=quay.io}/ncigdc/indexd"
+BASE_CONTAINER_VERSION=1.2.0  # upgrade later
 
-# avoid installing git
-COMMIT=$(git rev-parse HEAD) && echo "COMMIT=\"${COMMIT}\"" >indexd/index/version_data.py
+IMAGE_NAME="${DOCKER_RELEASE_REGISTRY:=quay.io}/ncigdc/${NAME}"
+REGISTRY="${BASE_CONTAINER_REGISTRY:=quay.io}"
+
+if [ -f 'VERSION.txt' ]; then
+  VERSION=$(cat VERSION.txt)
+else
+  VERSION=$(python -m setuptools_scm)
+fi
 
 # setup active branch name, default to using git if build is happening on local
 if [ ${TRAVIS_BRANCH+x} ]; then
   GIT_BRANCH=$TRAVIS_BRANCH;
 elif [ ${GITLAB_CI+x} ]; then
-  GIT_BRANCH=${CI_COMMIT_REF_SLUG}
+  GIT_BRANCH=${CI_COMMIT_REF_NAME};
 else
   GIT_BRANCH=$(git symbolic-ref --short -q HEAD);
 fi
 
 # replace slashes with underscore
 GIT_BRANCH=${GIT_BRANCH/\//_}
 
-VERSION=$(cat VERSION.txt)
-echo "$VERSION"
+# avoid installing git
+COMMIT=$(git rev-parse HEAD)
+
+echo "COMMIT=\"${COMMIT}\"" > indexd/index/version_data.py
 
 BUILD_COMMAND=(build \
   --label org.opencontainers.image.version="${VERSION}" \
   --label org.opencontainers.image.created="$(date -Iseconds)" \
-  --label org.opencontainers.image.revision="$(git rev-parse HEAD)" \
-  --label org.opencontainers.ref.name="indexd:${GIT_BRANCH}" \
-  --ssh default -t "$IMAGE_NAME:$GIT_BRANCH")
+  --label org.opencontainers.image.revision="${COMMIT}" \
+  --label org.opencontainers.ref.name="${NAME}:${GIT_BRANCH}" \
+  --build-arg REGISTRY="${REGISTRY%\/ncigdc}" \
+  --build-arg BASE_VERSION="${BASE_CONTAINER_VERSION:=1.2.0}" \
+  --build-arg PIP_INDEX_URL \
+  --build-arg REQUIREMENTS_GDC_LIBRARIES_FILE \
+  -t "$IMAGE_NAME:$GIT_BRANCH" \
+  -t "$IMAGE_NAME:$COMMIT"
+)
```

Comment:
Here's part of my confusion about labels added to the img. Also, outside the context of this PR.

We have Dockerfile and docker.sh and the 2 are intertwined ... I think we should create a jira and consolidate what's responsible for what and where it's done (gitlab template, base-containers, etc.). Maybe docker img builder in gitlab templates should have them both.

Reply:
I do think we should discuss this. 
But not some change should be made in this ticket.